### PR TITLE
Add nullable to return type for Serializable::serialize

### PR DESCRIPTION
**Pull request by mvorisek**
https://3v4l.org/JlkBD

### CODE REVIEW

**Comment by cmb69** [comment]
The [stubs](https://heap.space/xref/PHP-8.0/Zend/zend_interfaces.stub.php?r=68195bd4#49) do disagree, so they would need to be fixed first. Please provide a respective PR to php-src.

**Comment by mvorisek** [comment]
done - https://github.com/php/php-src/pull/6645

one question, what this the process behind updating the doc? When the stub is changed, is the doc updated semi/automatically?

**Comment by cmb69** [comment]
No, the docs have to be updated manually, but gen_stub.php can be used to create new methodsynopses, see https://github.com/php/doc-en/pulls?q=is%3Apr+is%3Aopen+label%3Amethodsynopses.

File: language/predefined/serializable/unserialize.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
   &reftitle.description;
   <methodsynopsis>
    <modifier>abstract</modifier> <modifier>public</modifier> <type>void</type><methodname>Serializable::unserialize</methodname>
-   <methodparam><type>string</type><parameter>serialized</parameter></methodparam>
+   <methodparam><type>?string</type><parameter>serialized</parameter></methodparam>
```
Comment (mvorisek) [inline language/predefined/serializable/unserialize.xml:14]:
```suggestion
   <methodparam><type>string</type><parameter>serialized</parameter></methodparam>
```

File: language/predefined/serializable/serialize.xml [hunk 2]
```diff
@@ -10,7 +10,7 @@
  <refsect1 role="description">
   &reftitle.description;
   <methodsynopsis>
-   <modifier>abstract</modifier> <modifier>public</modifier> <type>string</type><methodname>Serializable::serialize</methodname>
+   <modifier>abstract</modifier> <modifier>public</modifier> <type>?string</type><methodname>Serializable::serialize</methodname>
```
Comment (mvorisek) [inline language/predefined/serializable/serialize.xml:13]:
```suggestion
   <modifier>abstract</modifier> <modifier>public</modifier> <type>string|null</type><methodname>Serializable::serialize</methodname>
```

**Pull request closed without merging**